src/bvals/cc/flux_correction_cc.cpp

```diff
@@ -41,16 +41,21 @@
 namespace parthenon {
 
 //----------------------------------------------------------------------------------------
-//! \fn void CellCenteredBoundaryVariable::SendFluxCorrection()
+//! \fn void CellCenteredBoundaryVariable::SendFluxCorrection(bool is_allocated)
 //  \brief Restrict, pack and send the surface flux to the coarse neighbor(s)
 
-void CellCenteredBoundaryVariable::SendFluxCorrection() {
+void CellCenteredBoundaryVariable::SendFluxCorrection(bool is_allocated) {
   std::shared_ptr<MeshBlock> pmb = GetBlockPointer();
   auto &coords = pmb->coords;
   const IndexDomain interior = IndexDomain::interior;
 
   for (int n = 0; n < pmb->pbval->nneighbor; n++) {
     NeighborBlock &nb = pmb->pbval->neighbor[n];
+    if ((nb.snb.rank == Globals::my_rank) &&
+        (!local_neighbor_allocated[n] || !is_allocated)) {
+      continue;
+    }
```

Comment:
Great question. For non-local neighbors, we always send (and receive) the MPI message. The neighbor then decides what to do with it. If the neighbor has the variable allocated, it reads the buffer, otherwise it ignores it (receiving flux corrections for an unallocated variable won't result in an allocation). I've improved this communication routine a bit by adding a flag that indicates if the sending block has the variable allocated or not, so that the receiving block can ignore the buffer (which seems to contain 0's but could contain garbage) received from an unallocated sender.